samples/tls_between_enclaves/AttestedTLSREADME.md

```diff
@@ -0,0 +1,153 @@
+ *
+ * @param[out] output_cert a pointer to buffer pointer
+ * @param[out] output_cert_size size of the buffer above
+ *
+ * @return OE_OK on success
+ */
+oe_result_t oe_generate_attestation_certificate(
+    const unsigned char* subject_name,
+    uint8_t* private_key,
+    size_t private_key_size,
+    uint8_t* public_key,
+    size_t public_key_size,
+    uint8_t** output_cert,
+    size_t* output_cert_size);
+```
+#### Authenticate peer certificate
+
+Upon receiving a certificate from the peer endpoint, a connecting party needs to perform peer certificate validation.
+
+In this feature, instead of using the TLS API's default authentication routine, which validates the certificate against a pre-determined CAs for authentication, an application needs to conduct "Extended custom certificate validation" inside the peer custom certificate verification callback (cert_verify_callback), which is supported by all the popular TLS APIs.
+
+```
+For example:
+
+    Mbedtls:
+            void mbedtls_ssl_conf_verify(
+                      mbedtls_ssl_config *conf,
+                      int(*f_vrfy)(void *, mbedtls_x509_crt *, int, uint32_t *)
+                      void *p_vrfy)
+
+    OpenSSL:
+            void SSL_CTX_set_verify(
+                      SSL_CTX *ctx, int mode,
+                      int (*verify_callback)(int, X509_STORE_CTX *))
+```
+##### Custom extended certificate validation
+
+The following four validation steps are performed inside the cert_verify_callback
+  1. Validate certificate
+     - Verify the signature on the self-signed certificate to ascertain that the attestation report is genuine and unmodified.
+  2. Validate the quote
+     - Extract this quote extension from the certificate
+     - Perform quote validation
+  3. Validate the connection between the certificate and  the quote:
+     - Compute the SHA256 hash of the certificate's public key and compare this against the report’s user data in the quote
+     - If they match, it proves the certificate was cryptographically tied to the quote because hash of the public key was used as report data during quote generation
+  4. Validate peer enclave's identity
+     - Validate the enclave’s identity (e.g., MRENCLAVE in SGX) against the expected list. This check ensures only the intended party is allowed to connect to.
+
+  A new OE API, oe_verify_attestation_certificate(), was added to perform step 1-3 and leaving step 4 to application for business logic, which can be done inside a caller-registered callback, enclave_identity_callback, a callback parameter to oe_verify_attestation_certificate() call.
+
+  A caller wants to fail cert_verify_callback with non-zero code if either certificate signature validation failed or unexpected TEE identity was found. This failure return will cause the TLS handshaking process to terminate immediately, thus preventing establishing connection with a unqualified connecting party.
```

Comment:
Change "a unqualified" to "an unqualified"

Reply:
addressed